app/controllers/support_interface/provider_users_controller.rb

```diff
@@ -84,6 +84,19 @@ def toggle_notifications
       redirect_to support_interface_provider_user_path(provider_user)
     end
 
+    def update_notifications
```

Comment:
This controller seems to be quickly expanding, and I think we could benefit from splitting it up into more specific controllers (and handle things in a more railsy way) e.g. `app/controllers/support_interface/provider_user/notifications_controller.rb` with an `update` action, and its own update route rather than plugging another action on this controller.